byron-proxy/src/lib/Ouroboros/Byron/Proxy/Main.hs

```diff
@@ -345,48 +349,49 @@ bbsGetBlockHeaders db onErr mLimit checkpoints mTip = do
     _ : _ -> bbsStreamBlocks db onErr (headerHash newestCheckpoint) $ \producer ->
       runConduit $
            mapOutput Pos.Chain.Block.getBlockHeader producer
-        .| consumer (fromMaybe maxBound mLimit) []
+        .| consumer 0 (fromMaybe maxBound mLimit) []
   where
-  consumer :: Word
+  consumer :: Word -- Limit
+           -> Word -- Counted so far
            -> [BlockHeader]
            -> ConduitT BlockHeader x IO (Maybe (NewestFirst NonEmpty BlockHeader))
-  consumer 0 _   = pure Nothing
-  consumer n acc = do
-    next <- await
-    case next of
-      Nothing        -> pure Nothing
-      Just blkHeader ->
-        if maybe False ((==) (headerHash blkHeader)) mTip
-        then pure $ Just $ NewestFirst $ blkHeader NE.:| acc
-        else consumer (n-1) (blkHeader : acc)
+  consumer limit 0 acc = do
+    -- Drop the first one.
```

Comment:
why? is this the infamous ebb?

Reply:
No, it's to keep with the de facto cardano-sl contract. The checkpoint itself is not included in the result. Comment on lines 334, 335 explains it.